diff --git a/src/android/app/build.gradle.kts b/src/android/app/build.gradle.kts
index e8d8141711..1cc27f2369 100644
--- a/src/android/app/build.gradle.kts
+++ b/src/android/app/build.gradle.kts
@@ -178,7 +178,12 @@ android {
                     "-DCMAKE_EXPORT_COMPILE_COMMANDS=ON",
                     "-DBUILD_TESTING=OFF",
                     "-DYUZU_TESTS=OFF",
-                    "-DDYNARMIC_TESTS=OFF"
+                    "-DDYNARMIC_TESTS=OFF",
+                    "-DCMAKE_BUILD_TYPE=Release",
+                    "-DCMAKE_C_FLAGS=-fprofile-use=$projectDir/../../../../pgo/android.profdata -Wno-profile-instr-unprofiled -Wno-profile-instr-out-of-date -Wno-backend-plugin -flto=thin",
+                    "-DCMAKE_CXX_FLAGS=-fprofile-use=$projectDir/../../../../pgo/android.profdata -Wno-profile-instr-unprofiled -Wno-profile-instr-out-of-date -Wno-backend-plugin -flto=thin",
+                    "-DCMAKE_SHARED_LINKER_FLAGS=-flto=thin",
+                    "-DCMAKE_EXE_LINKER_FLAGS=-flto=thin"
                 )
 
                 abiFilters("arm64-v8a")
diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index 62dab070e3..497cc80482 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -1184,7 +1184,7 @@ else()
         -Wno-cast-function-type
 
         $<$<CXX_COMPILER_ID:Clang>:-fsized-deallocation>
-        $<$<CXX_COMPILER_ID:Clang>:-Wno-cast-function-type-mismatch>
+        $<$<CXX_COMPILER_ID:Clang>:-Wno-cast-function-type-strict>
     )
 endif()
 
