diff --git a/src/core/reporter.cpp b/src/core/reporter.cpp
index 1723636811..d4cd77c0fd 100644
--- a/src/core/reporter.cpp
+++ b/src/core/reporter.cpp
@@ -6,6 +6,7 @@
 
 #include <ctime>
 #include <fstream>
+#include <sstream>
 #include <iomanip>
 
 #include <fmt/chrono.h>
diff --git a/src/video_core/renderer_vulkan/vk_texture_cache.cpp b/src/video_core/renderer_vulkan/vk_texture_cache.cpp
index 466be26577..7f3c3f2a51 100644
--- a/src/video_core/renderer_vulkan/vk_texture_cache.cpp
+++ b/src/video_core/renderer_vulkan/vk_texture_cache.cpp
@@ -1378,8 +1378,8 @@ void TextureCacheRuntime::CopyImage(Image& dst, Image& src,
     // these images that aren't size-compatible
     if (BytesPerBlock(src.info.format) != BytesPerBlock(dst.info.format)) {
         auto oneCopy = VideoCommon::ImageCopy{
-            .src_offset = VideoCommon::Offset3D(0, 0, 0),
-            .dst_offset = VideoCommon::Offset3D(0, 0, 0),
+            .src_offset = VideoCommon::Offset3D{0, 0, 0},
+            .dst_offset = VideoCommon::Offset3D{0, 0, 0},
             .extent = dst.info.size
         };
         return ReinterpretImage(dst, src, std::span{&oneCopy, 1});
