diff --git a/CMakeLists.txt b/CMakeLists.txt
index dacbc73685..4e95e9caa8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -486,6 +486,7 @@ if (YUZU_USE_CPM)
         HASH 3037007f990040fe32573b46f9bef8762fd5dbeeb07ffffcbfeba51ec98167edae39bb9c87f9299efcd61c4e467c5e84f7c19f0df7799bc1fc04864a278792ee
         VERSION 1.5
         SOURCE_SUBDIR build/cmake
+        BUNDLED_PACKAGE ON
         OPTIONS
             "ZSTD_BUILD_SHARED OFF"
         EXCLUDE_FROM_ALL ON
diff --git a/externals/CMakeLists.txt b/externals/CMakeLists.txt
index 64592a8855..17164c8e88 100644
--- a/externals/CMakeLists.txt
+++ b/externals/CMakeLists.txt
@@ -102,6 +102,7 @@ AddPackage(
     REPO "KhronosGroup/SPIRV-Headers"
     SHA 4e209d3d7e
     HASH f48bbe18341ed55ea0fe280dbbbc0a44bf222278de6e716e143ca1e95ca320b06d4d23d6583fbf8d03e1428f3dac8fa00e5b82ddcd6b425e6236d85af09550a4
+    BUNDLED_PACKAGE ON
 )
 
 AddPackage(
@@ -189,6 +190,7 @@ if (YUZU_USE_EXTERNAL_VULKAN_SPIRV_TOOLS)
         REPO "KhronosGroup/SPIRV-Tools"
         SHA 40eb301f32
         HASH 58d0fb1047d69373cf24c73e6f78c73a72a6cca3b4df1d9f083b9dcc0962745ef154abf3dbe9b3623b835be20c6ec769431cf11733349f45e7568b3525f707aa
+        BUNDLED_PACKAGE ON
         OPTIONS
             "SPIRV_SKIP_EXECUTABLES ON"
     )
diff --git a/src/video_core/renderer_vulkan/vk_texture_cache.cpp b/src/video_core/renderer_vulkan/vk_texture_cache.cpp
index ba58060d20..01de9e179a 100644
--- a/src/video_core/renderer_vulkan/vk_texture_cache.cpp
+++ b/src/video_core/renderer_vulkan/vk_texture_cache.cpp
@@ -1378,8 +1378,8 @@ void TextureCacheRuntime::CopyImage(Image& dst, Image& src,
     // these images that aren't size-compatible
     if (BytesPerBlock(src.info.format) != BytesPerBlock(dst.info.format)) {
         auto oneCopy = VideoCommon::ImageCopy{
-            .src_offset = VideoCommon::Offset3D(0, 0, 0),
-            .dst_offset = VideoCommon::Offset3D(0, 0, 0),
+            .src_offset = VideoCommon::Offset3D{0, 0, 0},
+            .dst_offset = VideoCommon::Offset3D{0, 0, 0},
             .extent = dst.info.size
         };
         return ReinterpretImage(dst, src, std::span{&oneCopy, 1});
