diff --git a/src/yuzu/update_checker.cpp b/src/yuzu/update_checker.cpp
index 76b436d1d1..e54eb8d7f8 100644
--- a/src/yuzu/update_checker.cpp
+++ b/src/yuzu/update_checker.cpp
@@ -58,7 +58,7 @@ std::optional<std::string> UpdateChecker::GetResponse(std::string url, std::stri
 
 std::optional<std::string> UpdateChecker::GetLatestRelease(bool include_prereleases)
 {
-    constexpr auto update_check_url = std::string{Common::g_build_auto_update_api};
+    const auto update_check_url = std::string{Common::g_build_auto_update_api};
     std::string update_check_path = fmt::format("/repos/{}", std::string{Common::g_build_auto_update_repo});
     try {
         if (include_prereleases) { // This can return either a prerelease or a stable release,
