diff --git a/CMakeModules/GenerateSCMRev.cmake b/CMakeModules/GenerateSCMRev.cmake
index 2d7081b7db..eb3794459b 100644
--- a/CMakeModules/GenerateSCMRev.cmake
+++ b/CMakeModules/GenerateSCMRev.cmake
@@ -8,19 +8,19 @@
 include(GetSCMRev)
 
 function(get_timestamp _var)
-    string(TIMESTAMP timestamp UTC)
+    string(TIMESTAMP timestamp "%Y-%m-%d" UTC)
     set(${_var} "${timestamp}" PARENT_SCOPE)
 endfunction()
 
 get_timestamp(BUILD_DATE)
+git_count(GIT_COUNT)
 
 if (DEFINED GIT_RELEASE)
   set(BUILD_VERSION "${GIT_TAG}")
   set(GIT_REFSPEC "${GIT_RELEASE}")
   set(IS_DEV_BUILD false)
 else()
-  string(SUBSTRING ${GIT_COMMIT} 0 10 BUILD_VERSION)
-  set(BUILD_VERSION "${BUILD_VERSION}-${GIT_REFSPEC}")
+  set(BUILD_VERSION "${BUILD_DATE}-${GIT_COUNT}")
   set(IS_DEV_BUILD true)
 endif()
 
diff --git a/CMakeModules/GetSCMRev.cmake b/CMakeModules/GetSCMRev.cmake
index ee5ce6a91c..b61b8a3531 100644
--- a/CMakeModules/GetSCMRev.cmake
+++ b/CMakeModules/GetSCMRev.cmake
@@ -8,6 +8,23 @@ function(trim var)
   set(${var} ${new} PARENT_SCOPE)
 endfunction()
 
+function(git_count _var)
+    execute_process(
+        COMMAND "${GIT_EXECUTABLE}" rev-list --count HEAD
+        WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
+        RESULT_VARIABLE res
+        OUTPUT_VARIABLE out
+        ERROR_QUIET
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+
+    if(NOT res EQUAL 0)
+        set(out "${out}-${res}-NOTFOUND")
+    endif()
+
+    set(${_var} "${out}" PARENT_SCOPE)
+endfunction()
+
 set(TAG_FILE ${CMAKE_SOURCE_DIR}/GIT-TAG)
 set(REF_FILE ${CMAKE_SOURCE_DIR}/GIT-REFSPEC)
 set(COMMIT_FILE ${CMAKE_SOURCE_DIR}/GIT-COMMIT)
diff --git a/src/yuzu/about_dialog.cpp b/src/yuzu/about_dialog.cpp
index b7c0cd58d5..9874093a90 100644
--- a/src/yuzu/about_dialog.cpp
+++ b/src/yuzu/about_dialog.cpp
@@ -18,7 +18,7 @@ AboutDialog::AboutDialog(QWidget* parent)
     static const std::string build_id = std::string{Common::g_build_id};
     static const std::string compiler = std::string{Common::g_compiler_id};
 
-    static const std::string yuzu_build = fmt::format("Eden | {} | {}", description, compiler);
+    static const std::string yuzu_build = fmt::format("Eden Nightly | {} | {}", description, compiler);
 
     const auto override_build = fmt::format(fmt::runtime(
                                                 std::string(Common::g_title_bar_format_idle)),
@@ -33,7 +33,7 @@ AboutDialog::AboutDialog(QWidget* parent)
     }
     ui->labelBuildInfo->setText(
         ui->labelBuildInfo->text().arg(QString::fromStdString(yuzu_build_version),
-                                       QString::fromUtf8(Common::g_build_date).left(10)));
+                                       QString::fromUtf8(Common::g_build_id)));
 }
 
 AboutDialog::~AboutDialog() = default;
diff --git a/src/yuzu/main.cpp b/src/yuzu/main.cpp
index 19162a4ab7..f8628e1aa1 100644
--- a/src/yuzu/main.cpp
+++ b/src/yuzu/main.cpp
@@ -4193,11 +4193,11 @@ void GMainWindow::OnEmulatorUpdateAvailable() {
     update_prompt.addButton(QMessageBox::Yes);
     update_prompt.addButton(QMessageBox::Ignore);
     update_prompt.setText(
-        tr("Update %1 for Eden is available.\nWould you like to download it?").arg(version_string));
+        tr("Update %1 for Eden Nightly is available.\nWould you like to download it?").arg(version_string));
     update_prompt.exec();
     if (update_prompt.button(QMessageBox::Yes) == update_prompt.clickedButton()) {
         QDesktopServices::openUrl(
-            QUrl(QString::fromStdString("https://github.com/eden-emulator/Releases/releases/tag/") +
+            QUrl(QString::fromStdString("https://github.com/pflyly/eden-nightly/releases/tag/") +
                  version_string));
     }
 }
@@ -4209,7 +4209,7 @@ void GMainWindow::UpdateWindowTitle(std::string_view title_name, std::string_vie
     static const std::string build_id = std::string{Common::g_build_id};
     static const std::string compiler = std::string{Common::g_compiler_id};
 
-    static const std::string yuzu_title = fmt::format("Eden | {} | {}", description, compiler);
+    static const std::string yuzu_title = fmt::format("Eden Nightly | {} | {}", description, compiler);
 
     const auto override_title =
         fmt::format(fmt::runtime(std::string(Common::g_title_bar_format_idle)), build_id);
diff --git a/src/yuzu/update_checker.cpp b/src/yuzu/update_checker.cpp
index 8291987d73..b8eb4c1e1f 100644
--- a/src/yuzu/update_checker.cpp
+++ b/src/yuzu/update_checker.cpp
@@ -55,7 +55,7 @@ std::optional<std::string> UpdateChecker::GetResponse(std::string url, std::stri
 std::optional<std::string> UpdateChecker::GetLatestRelease(bool include_prereleases)
 {
     constexpr auto update_check_url = "http://api.github.com";
-    std::string update_check_path = "/repos/eden-emulator/Releases";
+    std::string update_check_path = "/repos/pflyly/eden-nightly";
     try {
         if (include_prereleases) { // This can return either a prerelease or a stable release,
             // whichever is more recent.
