diff --git a/CMakeLists.txt b/CMakeLists.txt
index dacbc73685..36d637276f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -392,18 +392,20 @@ if (YUZU_USE_CPM)
     endif()
 
     # boost
-    set(BOOST_INCLUDE_LIBRARIES algorithm icl pool container heap asio headers process filesystem crc variant)
+    set(BOOST_INCLUDE_LIBRARIES algorithm icl pool container heap asio headers process filesystem crc variant context)
     AddPackage(
         NAME Boost
         REPO boostorg/boost
         TAG boost-1.88.0
         ARTIFACT boost-1.88.0-cmake.7z
-
         HASH e5b049e5b61964480ca816395f63f95621e66cb9bcf616a8b10e441e0e69f129e22443acb11e77bc1e8170f8e4171b9b7719891efc43699782bfcd4b3a365f01
-
         GIT_VERSION 1.88.0
         VERSION 1.57
         EXCLUDE_FROM_ALL ON
+        OPTIONS
+            "BOOST_USE_PREBUILT OFF"
+            "BOOST_CONTEXT_ASM OFF"
+            "BOOST_B2_OPTIONS=address-model=64 architecture=arm target-os=windows toolset=msvc"
     )
 
     # really annoying thing where boost::headers doesn't work with cpm
diff --git a/CMakeModules/DownloadExternals.cmake b/CMakeModules/DownloadExternals.cmake
index 642d7d0df1..54dbf7bac9 100644
--- a/CMakeModules/DownloadExternals.cmake
+++ b/CMakeModules/DownloadExternals.cmake
@@ -7,14 +7,14 @@
 #   prefix_var: name of a variable which will be set with the path to the extracted contents
 set(CURRENT_MODULE_DIR ${CMAKE_CURRENT_LIST_DIR})
 function(download_bundled_external remote_path lib_name cpm_key prefix_var version)
-    set(package_base_url "https://github.com/eden-emulator/")
+    set(package_base_url "https://github.com/pflyly/eden-nightly/")
     set(package_repo "no_platform")
     set(package_extension "no_platform")
 
     if (WIN32 OR FORCE_WIN_ARCHIVES)
         set(CACHE_KEY "windows")
-        set(package_repo "ext-windows-bin/raw/master/")
-        set(package_extension ".7z")
+        set(package_repo "raw/refs/heads/main/")
+        set(package_extension ".zip")
     elseif (${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
         set(CACHE_KEY "linux")
         set(package_repo "ext-linux-bin/raw/master/")
diff --git a/src/core/arm/dynarmic/dynarmic_cp15.cpp b/src/core/arm/dynarmic/dynarmic_cp15.cpp
index c663adda19..0d5e5912ae 100644
--- a/src/core/arm/dynarmic/dynarmic_cp15.cpp
+++ b/src/core/arm/dynarmic/dynarmic_cp15.cpp
@@ -58,6 +58,8 @@ CallbackOrAccessOneWord DynarmicCP15::CompileSendOneWord(bool two, unsigned opc1
                     _mm_lfence();
 #elif defined(ARCHITECTURE_x86_64)
                     asm volatile("mfence\n\tlfence\n\t" : : : "memory");
+#elif defined(_MSC_VER) && defined(ARCHITECTURE_arm64)
+                    _Memory_barrier();
 #elif defined(ARCHITECTURE_arm64)
                     asm volatile("dsb sy\n\t" : : : "memory");
 #else
@@ -75,6 +77,8 @@ CallbackOrAccessOneWord DynarmicCP15::CompileSendOneWord(bool two, unsigned opc1
                     _mm_mfence();
 #elif defined(ARCHITECTURE_x86_64)
                     asm volatile("mfence\n\t" : : : "memory");
+#elif defined(_MSC_VER) && defined(ARCHITECTURE_arm64)
+                    _Memory_barrier();
 #elif defined(ARCHITECTURE_arm64)
                     asm volatile("dmb sy\n\t" : : : "memory");
 #else
